src/test/scala/parsing/extractHeadersTests.scala

```diff
@@ -0,0 +1,67 @@
+package parsing
+
+
+import models.{IteratorType, ParserInput, SeqType}
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers._
+import parsing.Parser.extractHeaders
+
+class extractHeadersTests extends AnyFlatSpec {
+
+
+
+  "iterator" should "return header" in{
+
+    val source: Iterator[String] = Iterator("Ballon dor","Karim Benzema")
+
+    val parseInputSource: ParserInput = ParserInput(
+      in = IteratorType(source),
+      csvDefinition = formats.CSV,
+      headers = Nil
+    )(Parser.encode)
+
+    val getHeader: ParserInput = extractHeaders(parseInputSource)
+    getHeader.headers shouldBe source.toList.take(1)
```

Comment:
here I recommend not using `take(1)` but actually testing directly against what you are expecting, now you're just testing `take(1)` against a `take(1)`

```suggestion
    getHeader.headers shouldBe Iterator("Ballon dor")
```

This applies to the other tests as well

Reply:
thanks for your recommendation, already done